src/result.ts

```diff
@@ -104,26 +104,24 @@ export function safeTry<
 // Promise<Result<T, E>>, not ResultAsync<T, E>, is used as the return type.
 export function safeTry<T, E>(
   body: () => AsyncGenerator<Err<never, E>, Result<T, E>>,
-): Promise<Result<T, E>>
+): ResultAsync<T, E>
 export function safeTry<
   YieldErr extends Err<never, unknown>,
   GeneratorReturnResult extends Result<unknown, unknown>
 >(
   body: () => AsyncGenerator<YieldErr, GeneratorReturnResult>,
-): Promise<
-  Result<
-    InferOkTypes<GeneratorReturnResult>,
-    InferErrTypes<YieldErr> | InferErrTypes<GeneratorReturnResult>
-  >
+): ResultAsync<
+  InferOkTypes<GeneratorReturnResult>,
+  InferErrTypes<YieldErr> | InferErrTypes<GeneratorReturnResult>
 >
 export function safeTry<T, E>(
```

Comment:
Could this become a set of function overloads?

I.e. one branch takes a generator, returns a Result. The other takes an async generator, returns ResultAsync.

Reply:
It's already like this

```ts
export function safeTry<T, E>(body: () => Generator<Err<never, E>, Result<T, E>>): Result<T, E>
```
and
```ts
export function safeTry<T, E>(
  body: () => AsyncGenerator<Err<never, E>, Result<T, E>>,
): ResultAsync<T, E>
```